src/main/kotlin/no/item/xp/plugin/util/XMLUtils.kt

```diff
@@ -0,0 +1,59 @@
+package no.item.xp.plugin.util
+
+import arrow.core.Either
+import org.w3c.dom.Document
+import org.w3c.dom.Node
+import org.w3c.dom.NodeList
+import javax.xml.xpath.XPathConstants
+import javax.xml.xpath.XPathExpression
+import javax.xml.xpath.XPathFactory
+
+val XPATH_FACTORY: XPathFactory = XPathFactory.newInstance()
+val XPATH_FORM_ELEMENT: XPathExpression = XPATH_FACTORY.newXPath().compile("//form/*[type]")
+val XPATH_MINIMUM_ELEMENT: XPathExpression = XPATH_FACTORY.newXPath().compile("//input/occurrences/@minimum")
+
+fun getFormElementChildren(doc: Document) = getNodesByXpath(doc, XPATH_FORM_ELEMENT)
+
+fun getMinimumOccurrencesList(node: Node) = getNodesByXpathAndNode(node, XPATH_MINIMUM_ELEMENT)
+
+fun getNodesByXpath(doc: Document, xpath: XPathExpression): Either<Throwable, Sequence<Node>> =
+  try {
+    Either.right(nodeListToSequence(xpath.evaluate(doc, XPathConstants.NODESET) as NodeList))
+  } catch(e: Exception) {
+    Either.left(e)
+  }
+
+fun getNodesByXpathAndNode(node: Node, xpath: XPathExpression): Either<Throwable, NodeList> =
+  try {
+    Either.right(xpath.evaluate(node, XPathConstants.NODESET) as NodeList)
+  } catch(e: Exception) {
+    Either.left(e)
+  }
+
+fun nodeListToSequence(nodeList: NodeList): Sequence<Node> =
+  (0..nodeList.length)
+    .map { nodeList.item(it) }
+    .asSequence<Node>()
+
+fun isOptional(node: Node): Boolean{
+  val nListMinimum: Either<Throwable, NodeList> = getMinimumOccurrencesList(node)
+  return nListMinimum.fold(
+    {false},
+    {convertMinimumValueToBoolean(it)}
+  )
+}
```

Comment:
I would not write to a variable if I didn't need to. Then I don't have to name it either.

```suggestion
fun isOptional(node: Node): Boolean =
  getMinimumOccurrencesList(node).fold(
    { false },
    { convertMinimumValueToBoolean(it) }
  )
```

Reply:
I had another suggestion further up regarding the signature of `getMinimumOccurrencesList` to return an `Option`.

You could end up with something even nicer like this:
```kotlin
fun isOptional(node: Node): Boolean =
  getMinimumOccurrencesList(node)
    .filter { isNotZero(it) }
    .getOrElse{ false }
```